### PR TITLE
[TIC-102] Fixed ability to purchase multiple ticket types for a showing

### PR DESCRIPTION
**Pull request by henrykaus**
### Description
Added the ability to purchase multiple ticket types for an event instance. This was done by adding the ticket type ID as an attribute in the CartItem, rather than _only_ the event instance ID. For this work, I added the ticket restrictions to the overall state of the program, and when adding a ticket to the cart, it checks against that table to ensure no more tickets are available to purchase than what is listed in the ticket restriction. If general admission, it only checks against the overall bank of tickets for that event instance. 

* This PR heavily edits the ticketingSlice file. Some are intentional changes, others are Prettier changes. 
* Added new V1 API to get all ticket restrictions

**Feel free to ask questions.**

### Risks
TicketingSlice was heavily edited so there may be unintended behavior that results from this.

I have yet to test with the pay what you can ticket type.

### Validation
Lots of manual testing in adding tickets in a variety of ways to the cart for bounds checking.

### Issue
[TIC-102](https://pdx-hkaus.atlassian.net/browse/TIC-102)

### Operating System
Windows 11

[TIC-102]: https://pdx-hkaus.atlassian.net/browse/TIC-102?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: client/src/components/Ticketing/event/TicketPicker.tsx [hunk 1]
```diff
@@ -400,7 +397,7 @@ const TicketPicker = (props: TicketPickerProps) => {
               select date
             </option>
             {tickets.map((t) => (
-              <option key={t.eventid} value={t.date.toString()}>
+              <option key={t.event_instance_id} value={t.date.toString()}>
```
Comment (henrykaus) [inline client/src/components/Ticketing/event/TicketPicker.tsx:400]: This fixes a "duplicate key" error that appeared in the console

File: client/src/components/Ticketing/ticketingmanager/showings/ShowingUpdated/EventShowingView.tsx [hunk 2]
```diff
@@ -21,7 +21,7 @@ export const EventShowingView = (props: EventInstanceViewProps) => {
 
   return (
     <div className={'bg-gray-300 rounded-xl p-2'}>
-            <div
+      <div
```
Comment (henrykaus) [inline client/src/components/Ticketing/ticketingmanager/showings/ShowingUpdated/EventShowingView.tsx:24]: Fixes docker eslint console error

File: client/src/components/Ticketing/cart/CartItem.tsx [hunk 3]
```diff
@@ -25,31 +15,26 @@ interface CartRowProps {
...
     } else {
       removeHandler(item.product_id);
     }
   };
 
   const handleIncrement = () => {
-    if (numAvailable && item.qty < numAvailable) {
-      dispatch(editItemQty({id: item.product_id, qty: item.qty + 1}));
-    }
+    dispatch(editItemQty({id: item.product_id, tickettypeId: item.typeID, qty: item.qty + 1}));
```
Comment (henrykaus) [inline client/src/components/Ticketing/cart/CartItem.tsx:37]: The amount validation was moved to ticketingSlice

File: client/src/components/Ticketing/event/TicketPicker.tsx [hunk 4]
```diff
@@ -237,7 +235,6 @@ const TicketPicker = (props: TicketPickerProps) => {
   }, []);
 
   const appDispatch = useAppDispatch();
-  const cartTicketCount = useAppSelector(selectCartTicketCount);
```
Comment (henrykaus) [inline client/src/components/Ticketing/event/TicketPicker.tsx:240]: `cartTicketCount` was not being used

**Comment by jonathantnguyen** [comment]
Based on testing, no issues revolving around correctly seeded data.

**Review by jonathantnguyen — APPROVED** [review]
Based on testing, no issues revolving around correctly seeded data.

**Pull request merged**